src/pathfinder/yapf/yapf_ship.cpp

```diff
@@ -173,7 +173,7 @@ class CYapfFollowShipT
 	}
 
 	/** Returns a random trackdir that can be reached from the current tile/trackdir, or INVALID_TRACK if none is available. */
-	static Trackdir GetRandomFollowUpTrackdir(const Ship *v, TileIndex tile, Trackdir dir, bool include_90_degree_turns)
+	static Trackdir GetRandomFollowUpTrackdir(const Ship *v, TileIndex &tile, Trackdir dir, bool include_90_degree_turns)
```

Comment:
TileIndex is a value type, but calling this function changes the provided tile. This is not visible from the calling site (in CreateRandomPath), which makes is a bit counter intuitive.

 Some alternative options:
- return a std::pair<TileIndex, Trackdir>
- Add a parameter, for example `TileIndex &new_tile`. Fill that tile with the new tile from the Trackfollower.

Reply:
I went with std::pair. I tried the 2nd alternative but I still fail at dealing with & and * stuff.